internal/cli/cli.go

```diff
@@ -186,27 +186,39 @@ func (c *RootCommand) runE(cmd *cobra.Command, args []string) error {
 
 	edc.SetEnabled(externalDNSEnabled)
 
-	var clientset *kubernetes.Clientset
-	var k8sInformerFactory k8sinformers.SharedInformerFactory
-	var nsInformer corev1informers.NamespaceInformer
-	var nsl corev1listers.NamespaceLister
-	if externalDNSEnabled {
-		clientset, err = kubernetes.NewForConfig(cfg)
-		if err != nil {
-			return err
-		}
+	clientset, err := kubernetes.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+
+	k8sInformerFactory := k8sinformers.NewSharedInformerFactoryWithOptions(clientset, time.Second*30)
 
-		k8sInformerFactory = k8sinformers.NewSharedInformerFactoryWithOptions(clientset, time.Second*30)
-		nsInformer = k8sInformerFactory.Core().V1().Namespaces()
+	nsInformer := k8sInformerFactory.Core().V1().Namespaces()
+
+	//need at least one listener func to populate the in memory cache
+	nsInformer.Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+		AddFunc: func(new interface{}) {},
+	})
 
-		//need at least one listener func to populate the in memory cache
-		nsInformer.Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
-			AddFunc: func(new interface{}) {},
-		})
+	coreV1Informer := k8sInformerFactory.Core().V1()
 
-		k8sInformerFactory.Start(wait.NeverStop)
-		k8sInformerFactory.WaitForCacheSync(wait.NeverStop)
-		nsl = nsInformer.Lister()
+	coreV1Informer.Services().Informer().AddEventHandler(k8scache.ResourceEventHandlerFuncs{
+		AddFunc: func(new interface{}) {},
+	})
+
+	k8sInformerFactory.Start(wait.NeverStop)
+	k8sInformerFactory.WaitForCacheSync(wait.NeverStop)
+
+	nsl := nsInformer.Lister()
+	serviceLister := coreV1Informer.Services().Lister()
+
+	if viper.GetBool("challenge-solver") {
+		cs := challengesolver.NewChallengeSolver(serviceLister, ic.NetworkingV1beta1(), cmc, glc)
```

Comment:
The `dryRun` option should be included in the `challengesolver` controller.

Reply:
There is no dry-run option implemented in the challengesolver so it wouldn't so very much. Dry run could perform all of the read operations. Do you want it to log the rendered patch or just GVK + name?

Reply:
Right, I am noticing now that it is wired up, that it should implement a dry-run option.  I believe it would only be used on the Virtual Service apply.

Reply:
> Do you want it to log the rendered patch or just GVK + name?

I think the GVK + name of the VS it would have applied would be good. Unless there is value of logging the full patch.

Reply:
GVK + name lets you know it would make the VS but not whether the solution is correct. The whole patch would let you validate that the settings are correct. 

I am gonna go with GVK + name because for now. We will certainly know if it isn't solving correctly.

Reply:
I went with just GVK namespace and name.